### ISSUE TITLE
want it to animate per second. How it can be done?

### ISSUE THREAD

**Issue by yogesh2209** [luispadron/UICircularProgressRing#5]
Function animate with duration is not working on second basis. It animates so quickly. Want it to animate per second. Want to implement it in quiz app as a countdown timer.
Please help.
And i know, Github is not for doubts. But, don't have any other option. It's damn urgent. Sorry. Please help.

**Comment by luispadron** [comment]
You're going to have to be more specific, I don't understand what you're asking. 
Provide code samples or be more clear in your question.

You can read the documentation [here](https://htmlpreview.github.io/?https://raw.githubusercontent.com/luispadron/UICircularProgressRing/master/docs/Classes/UICircularProgressRingView.html)

There you should see that to change the animation duration, you have to call `setProgress(value:animationDuration:completion:)` where `animationDuration` is the amount of seconds you would like to animate for.

**Comment by yogesh2209** [comment]
function animate with duration = 1.0 is not working on seconds basis.
I have tried with duration = 1, 10, 20, 30 and so on. Nothing works.
I just want it to animate per second i.e on seconds basis. So, for that, what should be the duration to animate it per second ?
FUNCTION: 
setProgress(value: 30, animationDuration: 1.0)

It is not animating on a second basis.
I compared it with stopwatch. It was not animating on per second basis.

**Comment by luispadron** [comment]
I'm not doing any custom animations with this view. I use the built in animation system from Apple.

Here is the code which calls the animation 

```swift
if event == "value" && self.animated {
            let animation = CABasicAnimation(keyPath: "value")
            animation.fromValue = self.presentation()?.value(forKey: "value")
            animation.timingFunction = CAMediaTimingFunction(name: animationStyle)
            animation.duration = animationDuration
            return animation
}
```

The animation duration is the value passed in when `setProgress` is called. Animation style can be change by assigning `animationStyle` for the ring view. The list of timing functions can be found [here](https://developer.apple.com/reference/quartzcore/camediatimingfunction/1668453-predefined_timing_functions).

Now as for the animation duration not being exact, I'm not sure why you're having that issue, I just tested it out with a stop watch as well and they seem to stop around the same time sent to the function (in seconds). 

Have you taken a look at the example project? If you don't mind posting your code, i can help further.

**Comment by yogesh2209** [comment]
kCAMediaTimingFunctionLinear should work on a linear basis i.e animation should work linearly without speeding fast or slow at any time. Right?

If i put total time = 30 seconds and compare it with stopwatch, it works fine but distribution of animation duration is not exact. It first slows, then speeds up and then becomes slow when reaching near the total. technically, it should be linearly distributed. No speed slow or quick. 
Got my point?
I am using linear timing function, still, it is not working.
ring.animationstyle = kCAMediaTimingFunctionLinear

I am doing it in your example project only. I will implement it in my app after i will do it on example project first.

**Comment by luispadron** [comment]
Huh thanks for bringing this to my attention. 
For some reason I wasn't updating the layers animationStyle.

Fixed in latest release, [1.1.3] (https://github.com/luispadron/UICircularProgressRing/releases/tag/v1.1.3) (run `pod update` for latest changes)

Will close issue unless you have something else.
